js_modules/dagit/packages/core/src/partitions/PartitionRangeInput.tsx

```diff
@@ -9,29 +9,40 @@ export const PartitionRangeInput: React.FC<{
   partitionNames: string[];
 }> = ({value, onChange, partitionNames}) => {
   const [valueString, setValueString] = React.useState('');
+  const partitionNameJSON = React.useMemo(() => JSON.stringify(partitionNames), [partitionNames]);
 
   React.useEffect(() => {
-    setValueString(partitionsToText(value, partitionNames));
-  }, [value, partitionNames]);
+    const partitionNameArr = JSON.parse(partitionNameJSON);
+    setValueString(partitionsToText(value, partitionNameArr));
+  }, [value, partitionNameJSON]);
 
   const placeholder = React.useMemo(() => {
     return placeholderForPartitions(partitionNames);
   }, [partitionNames]);
 
+  const tryCommit = (e: React.SyntheticEvent<HTMLInputElement>) => {
+    try {
+      onChange(textToPartitions(valueString, partitionNames));
+    } catch (err: any) {
+      e.preventDefault();
+      showCustomAlert({body: err.message});
+    }
+  };
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
```

Comment:
I think elsewhere in the app we use `Enter` or `Return` - want to do both here too? (I'm actually not sure which the return key is to be entirely honest, we may not need to do both anywhere.)

Reply:
I think we don't have to worry about `Return`, and `Enter` is sufficient.